api/src/main/java/com/inrupt/client/ClientHttpException.java

```diff
@@ -0,0 +1,53 @@
+package com.inrupt.client;
+
+import com.inrupt.client.spi.JsonService;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+public class ClientHttpException extends InruptClientException {
+    private ProblemDetails problemDetails;
+
+    /**
+     * Create a ClientHttpException.
+     * @param message the exception message
+     */
+    public ClientHttpException(ProblemDetails problemDetails, String message) {
+        super(message);
+        this.problemDetails = problemDetails;
+    }
+
+    /**
+     * Create a ClientHttpException.
+     * @param message the exception message
+     * @param cause a wrapped exception cause
+     */
+    public ClientHttpException(ProblemDetails problemDetails, String message, Exception cause) {
+        super(message, cause);
+        this.problemDetails = problemDetails;
+    }
+
+    public ProblemDetails getProblemDetails() {
+        return this.problemDetails;
+    }
+
+    private static ClientHttpException buildDefaultException(String message, int code, String status) {
+        return new ClientHttpException(
+                ProblemDetails.fromDefaultResponse(code, status),
+                message
+        );
+    }
+
+    public static ClientHttpException fromResponse(String message, int code, String status, Headers headers, InputStream body, JsonService jsonService) {
```

Comment:
This method should be removed

Reply:
This has been moved to a private method of `SolidClient` with 72018e0c54cd8902d2709f775ae4766da65ed616.